### DISCUSSION TITLE
[Dúvida]Devo estudar UX/UI Design junto com HTML, CSS e JavaScript?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#932] [Discussão]
Bem, estou começando e acabei a grade de HTML e vou começar CSS e depois partir pra JavaScript. É importante eu estudar UX/UI Design junto com o estudo front-end  focado na parte de programação ou devo estudar mais pra frente? E quais cursos de UX/UI Design vocês recomendariam?

**Comment by simaodeveloper** [comment]
Bruno eu recomendo você fortificar sua base para realizar o trabalho de um **Front-End Junior** primeiro:

- HTML, CSS, Javascript e jQuery;
- Básico de VueJS ou React ou Angular; (Muitas agências não pedem, é um diferencial);
- Baixe templates free de PSDs na internet para fixar o conhecimento obtido;
- Aprenda a utilizar plugins, validar formulário, slides, draggable; 
- Criar animações com CSS e JS (Veja TweenMax);
- Desenvolva sites responsivos (mobile, tablet e desktop);
- Estudo framework CSS como o Bootstrap (recomendo), Foundation ou SemanticUI;
  
Depois sim recomendo fazer esse curso de  UI/UX, pois você pode trabalhar em uma empresa que necessite muita comunicação com o time de criação, esse curso é praticamente uma especialização para designers que querem focar em desenvolvimento de interfaces (estou fazendo e curtindo):

- https://www.udemy.com/design-de-interface

E por último estudar back-end, sugiro PHP pela demanda de projetos que há no mercado para freelas ou qualquer linguagem que você goste e consiga desenvolver uma aplicação web, assim você vai entender como uma aplicação web funciona num todo, sugiro o canal e cursos do Brad Traversy:

- Canal: https://www.youtube.com/user/TechGuyWeb
- Cursos: https://www.udemy.com/user/brad-traversy/

O importante é você ter uma base geral, mas focar em algo, o front-end é muito mais do que eu disse acima, pois há mais bibliotecas, frameworks, ferramentas a se aprender, mas com o tempo e com a ajuda dos amigos, tanto da comunidade ou do futuro time que você trabalhar, você vai entendendo melhor o que precisa aprender e por em prática.

Espero ter ajudado!

See ya!!!

**Reply by davsdev** [reply]
Chegando aqui depois de uns anos, depois de consultar uma dúvida de carreira no Google. Sou designer e tenho bom fundamento em UX e UI, estou aprendendo programação em javascript, ainda nos fundamentos. Você segue com essa recomendação de base para front-end junior?

É uma boa, manter praticando todo conteúdo (UI e Front-End), no sentido de me destacar no mercado, ou devo deixar o antigo de lado e focar só em programação? Valeu \o

**Comment by ghost** [comment]
Opa, desculpa ter duplicado. Deu um erro quando fui criar essa e criei novamente.  Então @simaodeveloper  eu comprei um curso no udemy de desenvolvimento web da imediaflix e to gostando bastante. Eu realmente vou seguir  essa linha de HTML, CSS, JavaScript e JQuery que é a primeira grade do curso,  acabei a parte de HTML já. Por mais que o primeiro projeto em HTML e CSS seja bem simples, gostei bastante da maneira que ele explicou as tags então vou continuar e fazer tudo. Tem HTML, CSS, JavaScript, JQuery, Bootstrap e SEO. Estava em duvida sobre animações em CSS que é algo que gosto de ver e vou dar uma pesquisada no que você indicou. (:

**Comment by ghost** [comment]
Ah sim, esse curso também tem o back-end e é aquele clássico "PHP, MySQL" etc... porém estou pensando e ir logo pra MongoDB e NodeJS.  Ou vale a pena fazer todo o conteúdo do curso?

**Comment by simaodeveloper** [comment]
@LFeh deu ruim ai, essa aqui tem que deixar aberta! a #933 que é a duplicada

**Comment by simaodeveloper** [comment]
@brunojokubauskas Vale você dar uma olhada nas vagas e ver o que é mais pedido.

Existe o [Backendbr/vagas](https://github.com/backend-br/vagas/issues) 

Nesse link você consegue ter uma ideia do que estão pedindo, assim, você consegue pesquisar sobre a empresa e em quais tipos de projetos estão trabalhando, isso vai te ajudar a descobrir a tecnologias que estão mais em alta.

**Comment by simaodeveloper** [comment]
@brunojokubauskas  não localizei nada da **imediaflix na udemy**, manda o link para eu ver, por gentileza.

**Comment by kelciocajueiro** [comment]
Estuda sim @brunojokubauskas, conhecimento nunca é demais.

**Comment by ghost** [comment]
@simaodeveloper  é esse que estou fazendo. https://www.udemy.com/curso-desenvolvedor-web-completo/learn/v4/overview na verdade é com o instrutor andré fontenelle e é bem completo até onde estou vendo (:
